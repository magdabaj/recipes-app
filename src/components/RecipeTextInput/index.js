/**
 *
 * RecipeTextInput
 *
 */

import React, { memo } from 'react';
import '../../containers/RecipesFormContainer/index.css';
// import PropTypes from 'prop-types';
// import styled from 'styled-components';

function RecipeTextInput({ type, value, name, onChange, placeholder, label, error }) {
    let wrapperClass = 'input';
    if (error && error.length > 0) {
        wrapperClass += ' input-error';
    }
    return (
        <div className={'recipe-input-container'}>
            <label className={'label'}>{label}</label>
            <div className={'input-data input-data-recipe'}>
                <input
                    type={type}
                    name={name}
                    value={value}
                    placeholder={placeholder}
                    onChange={onChange}
                    className={wrapperClass}
                />
                {error && <div className={'error-message'}>{error}</div>}
            </div>
        </div>
    );
}

RecipeTextInput.propTypes = {};

export default memo(RecipeTextInput);
