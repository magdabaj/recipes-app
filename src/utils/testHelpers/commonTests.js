const commonTests = (render, useFirstChildAsSnapshot = true) => {
    it('does not log errors in console', () => {
        const spy = jest.spyOn(global.console, 'error')
        render()
        expect(spy).not.toHaveBeenCalled()
    })

    it('should render and match the snapshot',  () => {
        const { container } = render()
        expect(container).not.toBeNull()
        if(useFirstChildAsSnapshot) expect(container.firstChild).not.toBeNull()
        expect(
            useFirstChildAsSnapshot ? container.firstChild : container,
        ).toMatchSnapshot()
    });
}

export default commonTests;