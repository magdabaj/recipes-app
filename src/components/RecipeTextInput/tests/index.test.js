import RecipeTextInput from "../index";
import {render, screen} from "@testing-library/react";
import React from "react";
import commonTests from "../../../utils/testHelpers/commonTests";

const renderRecipeTextInput = (error = '') =>
    render(
        <RecipeTextInput
            type={'test'}
            value={'test'}
            name={'test'}
            onChange={jest.fn()}
            placeholder={'test'}
            label={'test'}
            error={error}
        />)

commonTests(renderRecipeTextInput)

test('shows no errors when they are not specified', () => {
        renderRecipeTextInput()
        const errorMessage = screen.queryByTestId('error-message')
        expect(errorMessage).not.toBeInTheDocument()
})

test('shows error when specified', () => {
        const {rerender} = renderRecipeTextInput()

        rerender(<RecipeTextInput
            type={'test'}
            value={'test'}
            name={'test'}
            onChange={jest.fn()}
            placeholder={'test'}
            label={'test'}
            error={'error'}
        />)

        const errorMessage = screen.getByTestId('error-message')
        const input = screen.getByLabelText(/test/i)
        expect(errorMessage).toBeInTheDocument()
        expect(errorMessage).toHaveTextContent('error')
        expect(input).toHaveClass('input input-error')
})

