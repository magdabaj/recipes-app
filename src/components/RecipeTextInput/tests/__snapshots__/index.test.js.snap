// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render and match the snapshot 1`] = `
<div
  class="recipe-input-container"
>
  <label
    class="label"
    for="test"
  >
    test
  </label>
  <div
    class="input-data input-data-recipe"
  >
    <input
      class="input"
      id="test"
      name="test"
      placeholder="test"
      type="test"
      value="test"
    />
    
  </div>
</div>
`;
